### ISSUE TITLE
Pyinstaller issue.

### ISSUE THREAD

**Issue by dillonvuong** [pyqtgraph/pyqtgraph#2298]
Hi I have the module not found error like others [[2179](https://github.com/pyqtgraph/pyqtgraph/issues/2179)] with pyinstaller. I could not figure out the answer from the other threads so I am making a new one. 

Here is my issue: 
![image](https://user-images.githubusercontent.com/39756756/167683079-a0456aec-c80e-45b7-a7f0-d54bd29ee57e.png)

I use the following spec file to create the pyqtgraph:

```python
`# -*- mode: python ; coding: utf-8 -*-
from PyInstaller.utils.hooks import collect_data_files
import os
import site

ROOT_PATH = os.getcwd()
PACKAGE_SITE= site.getusersitepackages()

print(ROOT_PATH)
print(PACKAGE_SITE)

datas = [(f'{ROOT_PATH}/config', './config'), (f'{ROOT_PATH}/assets', './assets'), (f'{ROOT_PATH}/log', './log')]
datas += collect_data_files('qt_material')

block_cipher = None


a = Analysis([f'{ ROOT_PATH }/main.py'],
             pathex=[ROOT_PATH, r"C:\Users\Beast\AppData\Roaming\Python\Python38\site-packages\pyqtgraph\graphicsItems\ViewBox\axisCtrlTemplate_pyside2.py"],
             binaries=[],
             datas=datas,
             hiddenimports=[],
             hookspath=[f'{PACKAGE_SITE}/pyupdater/hooks'],
             hooksconfig={},
             runtime_hooks=[],
             excludes=[],
             win_no_prefer_redirects=False,
             win_private_assemblies=False,
             cipher=block_cipher,
             noarchive=False)
pyz = PYZ(a.pure, a.zipped_data,
             cipher=block_cipher)

exe = EXE(pyz,
          a.scripts,
          [],
          exclude_binaries=True,
          name='win',
          debug=False,
          bootloader_ignore_signals=False,
          strip=False,
          upx=True,
          console=True,
          disable_windowed_traceback=False,
          target_arch=None,
          codesign_identity=None,
          embed_manifest=False,
          entitlements_file=None )
coll = COLLECT(exe,
               a.binaries,
               a.zipfiles,
               a.datas,
               strip=False,
               upx=True,
               upx_exclude=[],
               name='win')`
```

**Comment by j9ac9k** [comment]
Hi @dillonvuong 

I updated your post to have syntax highlighting for the spec file.

Can you provide the output of `pip list` please?

**Comment by dillonvuong** [comment]
@j9ac9k Thank you! Here is my pip list

```
Package                   Version
------------------------- ---------
altgraph                  0.17.2
appdirs                   1.4.4
attrs                     21.4.0
Automat                   20.2.0
boto3                     1.21.35
botocore                  1.24.35
bsdiff4                   1.2.2
certifi                   2021.10.8
cffi                      1.15.0
chardet                   4.0.0
constantly                15.1.0
cycler                    0.10.0
dsdev-utils               1.0.5
event-bus                 1.0.2
future                    0.18.2
hyperlink                 21.0.0
idna                      3.3
incremental               21.3.0
Jinja2                    2.11.3
jmespath                  1.0.0
joblib                    1.0.1
kiwisolver                1.3.1
MarkupSafe                1.1.1
mat4py                    0.5.0
matplotlib                3.3.4
mpv                       0.1
nidaqmx                   0.5.7
numpy                     1.20.1
pandas                    1.2.3
pbr                       5.8.1
pefile                    2021.9.3
Pillow                    8.1.0
pip                       22.0.4
shiboken2                 5.15.2
shiboken6                 6.0.1
six                       1.15.0
spinnaker-python          2.3.0.77
stevedore                 3.5.0
threadpoolctl             2.1.0
Twisted                   22.2.0
twisted-iocpsupport       1.0.2
typing_extensions         4.1.1
urllib3                   1.26.9
wheel                     0.37.1
zope.interface            5.4.0
```

**Comment by dillonvuong** [comment]
Ah sorry, I am using a venv with PyCharm

**Comment by dillonvuong** [comment]
Yes this is the requirements.txt and from what I understand it was created a while back with pip3 freeze? It might be outdated. How may I go about recreating a correct environment? 
```
cycler==0.10.0
event-bus==1.0.2
Jinja2==2.11.3
joblib==1.0.1
kiwisolver==1.3.1
MarkupSafe==1.1.1
matplotlib==3.3.4
mpv==0.1
numpy==1.20.1
pandas==1.2.3
Pillow==8.1.0
pyinstaller~=4.3
pymodbus~=2.5.1
pyparsing==2.4.7
PyQt5-sip==12.8.1
PyQt5-stubs==5.14.2.2
pyqtgraph~=0.12.1
pyserial==3.5
PySide2==5.15.2
python-dateutil==2.8.1
python-mpv==0.5.2
pytz==2021.1
qt-material==2.8.4
qtwidgets==0.18
scikit-learn==0.24.1
scipy==1.6.1
shiboken2==5.15.2
shiboken6==6.0.1
six==1.15.0
threadpoolctl==2.1.0
pymodbus3~=1.0.0
pyyaml
pyupdater~=4.0
pyupdater-s3-plugin~=4.1.2
dsdev-utils==1.0.5
sentry-sdk~=1.5.7
nidaqmx
mat4py

./spinnaker_python-2.3.0.77-cp38-cp38-win_amd64.whl
```

**Comment by j9ac9k** [comment]
ahh yeah, this does appear to have been created with `pip freeze` as it seems to have just dumped everything in the environment into the `requirements.txt` file.  In terms of recreating the  correct environment, that's a bit tough to do if your `requirements.txt` is just the output of `pip freeze`, so hard for me to say if it's "correct", the environment is as described it would appear.

If I was the one debugging this, I would create a new virtual environment with the python version I want to use, and try running that application from source, and install dependencies one at a time until the application runs successfully.

If you're also feeling adventurous; you can try installing pyqtgraph based on this current draft PR, and see if that fixes your issue

```bash
python -m pip uninstall pyqtgraph
python -m pip install git~https://github.com/pijyoi/pyqtgraph.git@generic_template
```

This installs pyqtgraph in the form of PR #2226, which @pijyoi was kind enough to submit (which seeks to do away w/ those template files).

**Comment by dillonvuong** [comment]
@j9ac9k I did as you said and deleted my venv folder and made a new environment. I installed packages one by one until the app ran. This fixed the problem for me. I am wondering why this is? Thank you for helping me!

**Comment by labbeast2020** [comment]
Oh also I had this warning. I am a little concerned but the app runs fine and the pyqtgraph shows correctly.

`Failed to collect submodules for 'pyqtgraph.canvas' because importing 'pyqtgraph.canvas' raised: ModuleNotFoundError: No module named 'CanvasManager'`

**Comment by dillonvuong** [comment]
Oh also I had this warning. I am a little concerned but the app runs fine and the pyqtgraph shows correctly.

`Failed to collect submodules for 'pyqtgraph.canvas' because importing 'pyqtgraph.canvas' raised: ModuleNotFoundError: No module named 'CanvasManager'`

**Comment by j9ac9k** [comment]
Thanks for the heads up on the canvas warning, should be harmless, that's not a portion of the library that is currently used very much (at all?).

In terms of "why" it started working, I'm not a great person to ask for this.  PyInstaller as far as I'm concerned works off of magic.  The task it does is hugely complex; and while I have used it plenty for my own work, I have used it enough to be very careful about how I do use it, as it's fairly sensitive.  Part of being careful with pyinstaller is not asking it to do any more than absolutely necessary, which means keeping dependencies to only what is strictly needed, and no more.  Of course, that doesn't answer the question of why was the error a pyqtgraph one, and to that effect I suspect there is likely something we _could_ do, but the only thing I can think of doing is already done here: https://github.com/pyinstaller/pyinstaller-hooks-contrib/blob/master/src/_pyinstaller_hooks_contrib/hooks/stdhooks/hook-pyqtgraph.py

Sorry I don't have a better answer for you.  Only thing I can say is if during your packaging you do come across the root cause of that issue, we would love to know about it!
